src/domain-engine.js

```diff
@@ -129,7 +129,7 @@ class DomainEngine {
     return next
   }
 
-  dispatch(state: Object, action: Action) {
+  dispatch(state: Object, action: Action): Object {
```

Comment:
And really hard to figure out within the context of `get` and `set`. I'm making a note to circle back on this.